pkg/reconciler/cluster/cluster.go

```diff
@@ -123,14 +132,48 @@ func (c *Controller) reconcile(ctx context.Context, cluster *v1alpha1.Cluster) e
 			cluster.Status.Conditions.SetReady(corev1.ConditionUnknown,
 				"SyncerInstalling",
 				"Installing syncer on cluster")
-		} else {
+		case SyncerModePush:
+			// TODO: specify more types
+			syncedResourceTypes := []string{"pods", "deployments"}
+
+			kubeConfig.CurrentContext = logicalCluster
+			bytes, err := clientcmd.Write(*kubeConfig)
+			if err != nil {
+				log.Printf("error writing kubeconfig for syncer: %v", err)
+				cluster.Status.Conditions.SetReady(corev1.ConditionFalse,
+					"ErrorInstallingSyncer",
+					fmt.Sprintf("Error installing syncer: %v", err))
+				return nil // Don't retry.
+			}
+			from, err := clientcmd.RESTConfigFromKubeConfig(bytes)
+			if err != nil {
+				log.Printf("error getting kcp kubeconfig: %v", err)
+				cluster.Status.Conditions.SetReady(corev1.ConditionFalse,
+					"ErrorStartingSyncer",
+					fmt.Sprintf("Error starting syncer: %v", err))
+				return nil // Don't retry.
+			}
+
+			to, err := clientcmd.RESTConfigFromKubeConfig([]byte(cluster.Spec.KubeConfig))
+			if err != nil {
+				log.Printf("error getting cluster kubeconfig: %v", err)
+				cluster.Status.Conditions.SetReady(corev1.ConditionFalse,
+					"ErrorStartingSyncer",
+					fmt.Sprintf("Error starting syncer: %v", err))
+				return nil // Don't retry.
+			}
+
+			s := syncer.New(to, from, syncedResourceTypes, cluster.Name)
+			c.syncers[cluster.Name] = s
+			s.Start(numSyncerThreads)
+		case SyncerModeNone:
```

Comment:
May I know why SyncerModeNone is needed? What is the use case for this mode?

Reply:
Mostly just for flexibility, to be able to run the syncer in isolation as a separate process. You could even run it remotely, it just needs two kubeconfigs and connectivity to kcp and the target cluster.

Over time we might decide it's not worth keeping this mode, but for now I didn't want to remove any behavior while adding this new mode.

Reply:
This would allow you keeping the logic of the cluster controller (that also manages the import of physical API resource definitions as CRDs in KCP) while replacing the default syncing with your own.